app/src/main/java/ch/epfl/sweng/erpa/services/UserAuthService.java

```diff
@@ -0,0 +1,43 @@
+package ch.epfl.sweng.erpa.services;
+
+import com.annimon.stream.Optional;
+
+import java.nio.charset.StandardCharsets;
+
+import javax.inject.Inject;
+
+import at.favre.lib.crypto.bcrypt.BCrypt;
+import ch.epfl.sweng.erpa.model.UserAuth;
+import ch.epfl.sweng.erpa.services.dummy.DummyRemoteServicesProvider;
+
+public class UserAuthService {
+
+    private RemoteServicesProvider rsp;
+
+    @Inject public UserAuthService() {
+        rsp = new DummyRemoteServicesProvider();
+    }
+
+    public Optional<UserAuth> getUserAuth(String username, String password) {
+        Optional<String> uid = rsp.getUidFromUsername(username);
+        if (uid.isPresent()) {
```

Comment:
Replace with `optional.map`

Reply:
Already tried but I cannot make it work because if the return statements